ast/rego_v1.go

```diff
@@ -0,0 +1,112 @@
+package ast
```

Comment:
We're now exposing internal compiler logic here. Should be put this in an `internal` package instead?

Reply:
I tried that, but since we also need to call this from the compiler which lives in the `ast` package which is heavily used from these functions, we get a circular import that fails compilation.
If there aren't clever ways around that, I suppose the alternative is to duplicate the functionality, so it can be used both in the compiler and in the formatter without making it public.

Perhaps the lesser evil is to eat the redundancy and not make this public .. I don't know 🤔. What do you think?

Reply:
I would say let's try to minimize the replication as much as we can. Exposing these methods does not seem like a good idea. Moving them to an `internal` package will require a lot of refactoring.